KWZP2019/KWZP2019/OshTrainingForm.cs

```diff
@@ -45,20 +47,76 @@ private void OshTrainingForm_FormClosed(object sender, FormClosedEventArgs e)
         private void OshTrainingForm_Load(object sender, EventArgs e)
         {
             List<Department> departmentList = db.Departments.ToList();
-            foreach(Department dep in departmentList)
+            foreach (Department dep in departmentList)
             {
                 domainUpDownDepartmentName.Items.Add(dep.DepartmentName);
             }
+            datePickerControlDate.Value = DateTime.Now;
         }
 
         private void btnShow_Click(object sender, EventArgs e)
         {
             dataGVOshTraining.DataSource = db.ViewOshTrainings.
                 Where(vOsh => vOsh.DepartmentName == domainUpDownDepartmentName.Text
-                && DbFunctions.AddDays(vOsh.TrainingDate, (int)vOsh.ValidityOfOshTraining * 365) <= DbFunctions.AddDays(DateTime.Now, 45)
+                && DbFunctions.AddDays(vOsh.TrainingDate, (int)vOsh.ValidityOfOshTraining) <= DbFunctions.AddDays(DateTime.Now, 45)
                 && vOsh.EndDate > DbFunctions.AddDays(DateTime.Now, 45))
-                .Select(osh => new { osh.EmployeeName, osh.EmployeeSurname, NextTrainingDate = DbFunctions.AddDays(osh.TrainingDate, (int)osh.ValidityOfOshTraining * 365) })
+                .Select(osh => new { Nr_pracownika = osh.IdEmployee, Imię_pracownika = osh.EmployeeName, Nazwisko_pracownika = osh.EmployeeSurname, Data_wygaśnięcia_szkolenia = DbFunctions.AddDays(osh.TrainingDate, (int)osh.ValidityOfOshTraining) })
                 .ToList();
         }
+
+        private void textBoxEmployeeId_TextChanged(object sender, EventArgs e)
+        {
+            Employee employee = db.Employees
+                .FirstOrDefault(check =>
```

Comment:
czemu `orDefault`?

Reply:
Ponieważ First() zwraca pierwszą znalezioną wartość odpowiadającą wymaganiom, jeżeli więc wpisując wartość ręcznie, pracownik korzystający z formularza wpisze wartość wykraczającą poza zakres tabeli, pojawia się error przerywający program.
FirstOrDefault() zwracając w takim przypadku default, obsługuje nieoczekiwaną wartość i program działa dalej.

Reply:
1. default czyli jaką wartosc? czy na pewno to jest oczekiwane dzialanie: np
- uzytkownik chcac wybrac Kowalskiego z id 34
- wpisze przez pomylke 344
- wyswietli mu sie Nowak bez zadnego komunikatu
- uzytkownik bedzie myslal ze widzi dane Kowalskiego i uzyje danych Nowaka
2. nie powinnismy dawac userowi mozliwosci wpisania wartosci wykraczajacej poza wartosc tabeli - z tego co widze wymagamy znajomosci ID od usera (czyli czegos czego nie moze znac!)
3. prosze zamienic ID na dropdown z imieniem i nazwiskiem

Reply:
1. Wartością domyślną dla odwołania i typów zerowalnych jest null. na podstawie https://docs.microsoft.com/pl-pl/dotnet/api/system.linq.enumerable.firstordefault?view=netframework-4.8
2. User czyli w przypadku formularza szkoleń bhp kierownik działu, zna numery pracowników oraz ich imiona i nazwiska dla których przeprowadzał szkolenie bhp ( na podstawie wygenerowanej w tym okienku listy takich pracowników ze swojego działu )
3. Co w przypadku jeśli w danym dziale są dwie osoby z takim samym imieniem i nazwiskiem? Wtedy taka lista imię + nazwisko nie jest wystarczająca

Reply:
ja to rozwiązałam tak że jest [Nazwisko] [Imię] " | NumerID :" [numer id]
![image](https://user-images.githubusercontent.com/48686261/57194012-c4fd4200-6f41-11e9-9fb7-c4b075401fb5.png)

Reply:
Dokonałam zmiany na DropDown, podobnie jak u Kingi, numer pracownika również jest w nim widoczny.
Tak jak napisał Bartek, w przypadku dwóch osób o takim samym imieniu i nazwisku, numer pomaga ich zidentyfikować.

Reply:
@Bartelix @kingak123 @minimilenka 
Szanowni Państwo, identyfikator bazy danych jest daleki od definicji numeru użytkownika. Jest to identyfikator czysto techniczny i jego znajomosc przez uzytkownika może prowadzic do incydentu bezpieczeństwa. Zatem kolejny raz upominam: w GUI nie prezentujemy identyfikatorów z bazy danych. Jesli potrzebny jest numer pracownika to proszę dodac taką kolumnę w tabeli. Takim numerem może byc np numer PESEL, ktory jest unikalny dla każdego obywatela RP.
@Bartelix jesli default zwroci null to jeszcze gorzej ponieważ może to doprowadzic do rzucenia wyjątkiem `NullReferenceException` i zatrzymania działania aplikacji (specjalny warunek aby sie przed tym ustrzec dodany jest w linii 73.

Zatem:
1. proszę usunac wyswietlanie a tymbardziej wymaganie od uzytkownika identyfikatorow bazy danych
2. prosze usunac `orDefault`